### PR TITLE
Feature/852 simple nav gyro

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-852
* **Review:** By commit 
* **Merge strategy:** Merge (no squash) 

## Description
Update simple nav to write out the accelerometer packet message (containing acceleration and rate information). This change allows for a user to set a cadence of gyro measurements per time step (100 measurements per time step for instance) and set gyro biases and standard deviations

## Verification
The unit test contains data to check the time output by the message (mean time for all the measurements) and the statistics on the noise

## Documentation
Updated given new message

## Future work
The module needs a healthy refactoring

### CODE REVIEW

File: src/simulation/navigation/simpleNav/simpleNav.rst [hunk 1]
```diff
@@ -38,6 +38,9 @@ not come from a noisy sensor but from a ground-based estimation process.
     * - scEphemOutMsg
       - :ref:`EphemerisMsgPayload`
       - spacecraft ephemeris output msg
+    * - scEphemOutMsg
+      - :ref:`accelDataOutMsg`
```
Comment (codyallard28) [inline src/simulation/navigation/simpleNav/simpleNav.rst:42]: accelDataOutMsg or imuDataOutMsg just going based on the comment description?

File: src/simulation/navigation/simpleNav/simpleNav.cpp [hunk 2]
```diff
@@ -206,6 +228,17 @@ void SimpleNav::computeErrors(uint64_t CurrentSimNanos)
     this->errorModel.setPropMatrix(localProp);
     this->errorModel.computeNextState();
     this->navErrors = this->errorModel.getCurrentState();
+
+    //! - Compute accelerometer errors
+    std::random_device rd;
```
Comment (codyallard28) [inline src/simulation/navigation/simpleNav/simpleNav.cpp:233]: rd haha

File: src/simulation/navigation/simpleNav/simpleNav.rst (hunk 1 above)
Comment (codyallard28) [reply src/simulation/navigation/simpleNav/simpleNav.rst:42]: Wait, it looks like this is just gyro data, so shouldn't this be gyroDataOutMsg?

**Review by codyallard28 — APPROVED** [review]
Would we ever use omega_Dot directly from a gyro? Isn't deltaTheta and omega coming from a gyro, and then we take omega and numerically differentiate to find omegaDot? Regardless, I approve.

File: src/simulation/navigation/simpleNav/simpleNav.rst (hunk 1 above)
Comment (thibaudteil) [reply src/simulation/navigation/simpleNav/simpleNav.rst:42]: The message was named accel Data from Scott. I just kept the name consistent

File: src/simulation/navigation/simpleNav/_UnitTest/test_simpleNav.py [hunk 3]
```diff
@@ -160,6 +171,52 @@ def simple_nav_sim(show_plots, noise=False, number_steps=1, sigma_test=3):
...
+    avg_accel = np.mean(accel_array, axis=0)
+    avg_time = np.mean(time_list)
+    std_rate = np.std(rate_array, axis=0)
+    std_accel = np.std(accel_array, axis=0)
+
+    np.testing.assert_allclose(avg_time,
+                               (1E9/simple_nav_object.gyroFrequencyPerSecond)*(simple_nav_object.numberOfGyroBuffers - 1)/2,
+                               atol=3*simple_nav_object.gyroStandardDeviation/np.sqrt(number_steps),
+                               equal_nan=False,
+                               err_msg="Middle of the time measurement failed",
```
Comment (patkenneally) [inline src/simulation/navigation/simpleNav/_UnitTest/test_simpleNav.py]: :large_orange_diamond: Let's edit these error messages to be more descriptive of what was expected. We can leave out any mention of failure as that is taken care of by the assertion apparatus.

File: src/simulation/navigation/simpleNav/simpleNav.h [hunk 4]
```diff
@@ -49,6 +49,14 @@ class SimpleNav: public SysModel {
     void writeOutputMessages(uint64_t Clock);
 
 public:
+    double gyroStandardDeviation=1E-5;    //!< Standard deviation for each rate component
```
Comment (patkenneally) [inline src/simulation/navigation/simpleNav/simpleNav.h:53]: :large_orange_diamond: Any value on making these setters/getters? I know the rest of the module doesn't have them but I can't immediately see a reason to not head in that direction. 🤔

File: src/simulation/navigation/simpleNav/simpleNav.cpp [hunk 5]
```diff
@@ -88,9 +89,10 @@ void SimpleNav::Reset(uint64_t CurrentSimNanos)
         bskLogger.bskLog(BSK_ERROR, "Your walkbounds vector  is not 18 elements. Quitting");
     }
     this->errorModel.setUpperBounds(this->walkBounds);
+    vSetZero(this->gyroErrors, 3 * MAX_ACC_BUF_PKT);
+    vSetZero(this->accelErrors, 3 * MAX_ACC_BUF_PKT);
 }
 
-
```
Comment (patkenneally) [inline src/simulation/navigation/simpleNav/simpleNav.cpp:93]: Rogue deletion

**Review by patkenneally — APPROVED** [review]
I just spotted some additional things worth considering. We could also come back later for a large module refactor.

File: src/simulation/navigation/simpleNav/simpleNav.h (hunk 4 above)
Comment (thibaudteil) [reply src/simulation/navigation/simpleNav/simpleNav.h:53]: should I clean it up and make setters/getters for everything?

File: src/simulation/navigation/simpleNav/_UnitTest/test_simpleNav.py (hunk 3 above)
Comment (thibaudteil) [reply src/simulation/navigation/simpleNav/_UnitTest/test_simpleNav.py]: done

File: src/simulation/navigation/simpleNav/simpleNav.h (hunk 4 above)
Comment (thibaudteil) [reply src/simulation/navigation/simpleNav/simpleNav.h:53]: We'll do this in a separate ticket

**Pull request merged**